### DISCUSSION TITLE
Airflow flower report inconsistent task status with airflow WebUI

### DISCUSSION THREAD

**Discussion by xiaolongz007** [apache/airflow#20782] [Q&A]
I am now using airflow with CeleryExecutor. I am using BashOperator, the situation is that when the shell script fails with a non-zero exit code, the WebUI correctly report it as failure but flower will report it as success. At the same time, If I manually mark the task as failed, flower still report the task success.

**Comment by potiuk** [comment]
I do not think you should expect flower tasks to "fail" in this case. They fail when something is wrong internally (error - exception with executing process etc.), but I believe from the point of view of flower, Airflow task is a success when it completes without an internal error of the worker - whether it returns failure or success (which is logical Airlflow task not Celery worker task) should be irrelevant for flower. It knows task "completed" and passed its status to Airflow (whether it is "failure" or "success" it does not matter), so it sees the tasks as successfullly completed.

There are just simply logically different "statuses" - similarly as in TCP/HTTP - you get "successful" TCP message that delivers HTTP 500 error message  - but it does not mean that the TCP message should be "failed"

Out of curiosity Why would you need to see status of Airflow task in Flower? Why do you think you need those two synchronized?

**Comment by xiaolongz007** [comment]
Sorry for the late reply and Thank you very much for saving me one day's
trial and error. The reason why I think they should be consistent is that
both of them have four task states and the states can be matched, so in my
subconsciousness, I really think flower should be able to track the task
states of airflow's logic task and display it in the WebUI. As a result, it
comes to me as wrong behavior. To solve it, I dive into the source code and
read something about the celeryExecutor, how the task state is updated, but
I was perplexed by the asynchronous code. But After your explanation, I
reread the code and I find it seems that the airflow task state is
calculated from the celery task which is exactly the opposite to what I
think before. Thank you again for your help, otherwise i will be still
trapped in the codebase for more time and it is really a good explanation
with http status code example and this is really a good lesson so that i
will ask myself first why i should think this way next time.
